dls-controls/malcolmjs
# Feature/layout delete #282

## Description

Hooked layout up so that deleting blocks or links by pressing the delete key sends appropriate Put to Malcolm server to make it so. Also added checks to prevent the layout responding to the delete key when it doesn't have focus and better handling for deleting multiple blocks.

## Testing instructions

Add a set up instructions describing how the reviewer should test the code
- Review code
- Check Travis build
- Review changes to test coverage
- npm start
- navigate to localhost:3000/PANDA/layout
- select a link by clicking on it
- check that the info panel opens
- delete the link by pressing delete or backspace
- move one of the remaining blocks and check that the deleted link doesn't re-appear on a server update.
- select a block by clicking on it
- click a widget in the newly opened child panel and then press the delete key to check that the layout does not respond to this.
- select multiple blocks by shift-clicking (leaving at least one block remaining)
- press delete or backspace
- check that the blocks disappear in one go
- move one of the remaining blocks and check that the deleted blocks don't re-appear on a server update.


## Agile board tracking

connect to #282 
closes #282
